### PR TITLE
Added gesv and test

### PR DESCRIPTION
**Pull request by stonecoldhughes**
The test for double precision float always works, but the test for single precision float version is off by tiny epsilon deviations - is it floating point representation limitations or is there an error I've missed?

### CODE REVIEW

**Comment by grahamlopez** [comment]
okay to test

File: src/lib_dispatch.cpp [hunk 1]
```diff
@@ -270,6 +270,35 @@ void gemm(char const *transa, char const *transb, int *m, int *n, int *k,
   }
 }
 
+template<typename P>
+void gesv(int *n, int *nrhs, P *A, int *lda, int *ipiv, P *B, int *ldb,
+          int *info)
+{
+  assert(n);
```
Comment (grahamlopez) [inline src/lib_dispatch.cpp]: can you make any stronger assertions for all of these?

File: src/lib_dispatch.cpp [hunk 2]
```diff
@@ -270,6 +270,35 @@ void gemm(char const *transa, char const *transb, int *m, int *n, int *k,
   }
 }
 
+template<typename P>
+void gesv(int *n, int *nrhs, P *A, int *lda, int *ipiv, P *B, int *ldb,
```
Comment (grahamlopez) [inline src/lib_dispatch.cpp]: can you remind me why we want to expose `gesv()` in lib_dispatch? and then if yes, why not also in fast_math?
Put these kinds of things in the PR description

File: src/lib_dispatch.hpp [hunk 3]
```diff
@@ -108,6 +118,10 @@ void gemm(char const *transa, char const *transb, int *m, int *n, int *k,
           P *alpha, P *A, int *lda, P *B, int *ldb, P *beta, P *C, int *ldc,
           resource const res = resource::host);
 
+template<typename P>
+void gesv(int *n, int *nrhs, P *A, int *lda, int *ipiv, P *B, int *ldb,
```
Comment (grahamlopez) [inline src/lib_dispatch.hpp]: need extern template decls as well

File: src/lib_dispatch_tests.cpp [hunk 4]
```diff
@@ -9,8 +9,7 @@
 //
 TEMPLATE_TEST_CASE("matrix-matrix multiply (lib_dispatch::gemm)",
                    "[lib_dispatch]", float, double, int)
-{
-  // clang-format off
+{ // clang-format off
```
Comment (grahamlopez) [inline src/lib_dispatch_tests.cpp]: restore this line to its original position

**Comment by grahamlopez** [comment]
@stonecoldhughes let's get this dealt with or closed, whichever you think is the right path.

**Comment by grahamlopez** [comment]
it looks like this might be obviated by #155

**Comment by bmcdanie** [comment]
gesv w/tests implemented in #155

**Pull request closed without merging**